viewing_party/party.py

```diff
@@ -1,23 +1,151 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title == None or genre == None or rating == None:
+        movie_data = None
+
+    else:
+        movie_data = {
+            'title': title,
+            'genre': genre,
+            'rating': rating
+            }
+
+    return movie_data
+
+
+def add_to_watched(user_data, movie):
+    user_data['watched'].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data['watchlist'].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, MOVIE_TITLE_1):
+    for movies in user_data.values():
+        for movie in movies:
+            if movie["title"] == MOVIE_TITLE_1:
+                add_movie = movies.pop(movies.index(movie))
+                user_data["watched"].append(add_movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    try:
+        total = 0
+        num_movies = 0
+
+        for movies in user_data.values():
+            for movie in movies:
+                num_movies += 1
+                total += movie['rating']
+        average = total / num_movies
+    except ZeroDivisionError:
+        average = 0.0
+
+    return average
+
+
+def get_most_watched_genre(user_data):
+    if user_data["watched"]:
+        most_popular_genre = []
+        for movies in user_data.values():
+            for movie in movies:
+                most_popular_genre.append(movie['genre'])
+        user_data = max(set(most_popular_genre), key = most_popular_genre.count)

```

Comment:
`user_data` is the name of our function parameter, so I recommend creating a new variable here with a name that describes the kind of value we will be returning to the user. 

From a readability perspective, we are currently overwriting the user_data variable with drastically different data than it was holding before. This can make it hard for someone unfamiliar with the code to reason about what that data should be representing and what type of data it might hold at any given time.